### ISSUE TITLE
VM crash while attempting to step over line in debugger

### ISSUE THREAD

**Issue by tvolkert** [dart-lang/sdk#28980]
I set a breakpoint in flutter_tools/src/android/android_sdk.dart line 70 and ran the tool in IntelliJ.  The breakpoint hit, but when I tried to step over the code, the VM crashed:

https://gist.github.com/tvolkert/db4b40b30a98384e99386652c4e9ce47

@rmacnak-google

**Comment by rmacnak-google** [comment]
@mhausner - Includes a context mismatch in async code

**Comment by mhausner** [comment]
@tvolkert could you please copy the source lines around the breakpoint that caused the crash?

A fresh checkout has this in android_sdk.dart:70

```
68   static AndroidSdk locateAndroidSdk() {
69     String androidHomeDir;
70     if (platform.environment.containsKey(kAndroidHome)) {
71       androidHomeDir = platform.environment[kAndroidHome];
72     } else if (platform.isLinux) {
```

**Comment by tvolkert** [comment]
Yep, that's correct - I had no local modifications.

I *was* running via --replay-from (unzipped the bug report attached to https://github.com/flutter/flutter/issues/8612, and ran `flutter/executable.dart` via the following args: `['doctor', '--replay-from=<path_to_unzipped_directory>']`

**Comment by mhausner** [comment]
I can repro this outside of IntelliJ and with my own dart executable. I also found that it doesn't happen with an older dart build, so it's a recent regression.

**Comment by mhausner** [comment]
It's a regression introduced by https://codereview.chromium.org/2646093006.

**Comment by fsc8000** [comment]
Could be the same issue as the one I tried to fix: Mapping source positions to context levels.

**Comment by mhausner** [comment]
Fix for review: https://codereview.chromium.org/2740783007/

**Comment by mhausner** [comment]
Fixed in d8ddd7766b2520e8cda889ec01930181068aad40.
